Tools/TestWebKitAPI/Tests/WTF/SynchronizedFixedQueue.cpp

```diff
@@ -123,6 +118,7 @@ class ToUpperConverter {
             return;
 
         m_lowerQueue->close();
+        m_upperQueue->close();
```

Comment:
Yes this will fix the issue but I think it is a little bit ugly. As you explained in the ChangeLog, the ProduceOnly problem happens because the main thread is blocked on
```
    m_produceCloseSemaphore.wait();
```
while the produceQueue() is blocked on 
```
    m_upperQueue->enqueue(toUpper(lower));
```
`m_upperQueue` is full since the `consumeQueue` is not running. Your solution fixes this issue by closing `m_upperQueue` so, `produceQueue()` can signal `m_produceCloseSemaphore`.

Alternatively I suggest the following:
1. Increase the size of `ToUpperConverter` in ProduceOnly to 8U. So no waiting in `produceQueue()`.
2. Change the type of `m_produceCount` and `m_consumeCount` to be `std::atomic<size_t>`
3. Change the return type of produceCount() and consumeCount() to be `const std::atomic<size_t>&`
4. Spin in `ProduceOnly` till `produceCount()` is equal to `count`.
5. Please Add these checks at the end of `ProduceOnly`
```
    EXPECT_EQ(converter.produceCount(), count);
    EXPECT_EQ(converter.consumeCount(), 0U);
```

Also in the `Limits` sub-test, please replace the `sleep_for()` with this spin:

```
    while (converter.consumeCount() < count)
        std::this_thread::sleep_for(std::chrono::milliseconds(1));
```

Reply:
Fixed. Thank you very much.